lib/gateway/pipelines.js

```diff
@@ -190,12 +191,16 @@ function validateGatewayConfig (config) {
   if (!gatewayConfig.pipelines) {
     if (gatewayConfig.pipeline) {
       gatewayConfig.pipelines = Array.isArray(gatewayConfig.pipeline) ? gatewayConfig.pipeline : [ gatewayConfig.pipeline ];
-    } else throw new ConfigurationError('No pipelines found');
+    } else {
+      gatewayConfig.pipeline = [];
```

Comment:
what if we merge line 191 to this in something like 
gatewayConfig.pipelines = gatewayConfig.pipelines || gatewayConfig.pipeline || [] ? 

it will reduce nesting, and later in the code  always use plural form

Reply:
and pipelines is not an array, it is an object 
name: pipelineDefinition